### DISCUSSION TITLE
We use this library in a network program.There is a  situation we have to overcome.

### DISCUSSION THREAD

**Discussion by helloword010** [nlohmann/json#2464] [Q&A] [answered]
we are a server  and receive the json string from another server. it is not a complete json string we received one time,but it is Orderly .that means we have to receive serval times and  we have to make sure it is  a full json. now we have use exception to make sure it is a full json. i don't think it is a good way.Is there a better way?

**Comment by gregmarr** [comment, marked as answer]
You can use this function to check before parsing, but then you basically have to parse every message at least twice.  Depending on why you are trying to avoid exceptions, that may be an acceptable trade-off.

    accept()
    check if the input is valid JSON
    Unlike the parse(InputType&&, const parser_callback_t,const bool)
    function, this function neither throws an exception in case of invalid JSON
    input (i.e., a parse error) nor creates diagnostic information.
